### DISCUSSION TITLE
The CustomResourceDefinition "kafkas.kafka.strimzi.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes

### DISCUSSION THREAD

**Discussion by XBeg9** [strimzi/strimzi-kafka-operator#4848] [Q&A] [answered]
Any ideas on how to fix this, looks like the IaC that I am using, tries to do `kubectl apply`, which is good for diffs.

Thanks

**Comment by scholzj** [comment, marked as answer]
You need to use `kubectl create` or `kubectl replace`. `kubectl apply` has a limitation to the size of the files for which it can be used. If you really want to use `kubectl apply`, you can remove the schema specifications from the CRDs manually to make them smaller (but you will miss the schemas and the validation).

**Reply by XBeg9** [reply]
Thanks @scholzj , that's what I am using so far, is it due to CRD v1? Can I delete old versions?

**Reply by scholzj** [reply]
Yes, it is related to the migration to CRD v1. We have now quite a lot of different Strimzi API versions in there. because of the migration and that makes them big. 

CRD v1 will be supported in the next release (I hope to start releasing it next week). That should have only the Strimzi `v1beta2` APIs and should work fine again with `kubectl apply`. We just have to somehow survive it through the 0.22 release which still needs all the Strimzi APIs :-/.

**Reply by XBeg9** [reply]
If I will use CRDs from the `0.20` version, will I break something? Just curious... the meantime, probably I will just downgrade to `0.20`. There is no way to install CRDs using IaC.

**Reply by scholzj** [reply]
Well, you have to do it one day if you don't want to be stuck on the old versions forever. I don't know what tooling are you using. But if it doesn't allow you to use `kubectl create` / `kubectl replace` you might need to deal with it sooner or later for example by removing the schemas as suggested before.

**Reply by XBeg9** [reply]
Anyway, thanks @scholzj for the awesome project. I just deleted my bitnami kafka cluster :)
